tests/core/CircularDependencyErrorTest.py

```diff
@@ -5,16 +5,16 @@
 
 class CircularDependencyErrorTest(unittest.TestCase):
 
-    def test_default_message(self):
+    def test_message(self):
         with self.assertRaises(CircularDependencyError) as cm:
-            raise CircularDependencyError
+            # test the default case (node is None)
+            raise CircularDependencyError()
```

Comment:
You can keep `raise CircularDependencyError`, this automatically invokes `__init__` without / with default arguments.